sdlib/dmd/thread.d

```diff
@@ -48,7 +66,14 @@ void __sd_gc_pre_suspend_hook(void* stackTop) {
 		 * pushed on it.
 		 */
 		import d.gc.tcache;
-		threadCache.clearTLSSegments();
+		auto tls = threadCache.clearTLSSegments();
+		if (tls.ptr !is null) {
+			// Arena needs a CachedExtentMap for freeing pages. Copy the
+			// threadCache to a temporary, so we don't mess with it.
+			auto emap = threadCache.emap;
```

Comment:
I don't think we want to guarantee that any intermediate state of the emap is a valid state.

Reply:
However, you can read the value of the base (in any case, it is not going to be used here) and the value of the emap, which never changes.

Reply:
To look at it a different way -- we have always been using the emap in this state in this call. So this is no *worse* than the current usage, and I've never seen it fail yet.

That doesn't mean you are wrong... In any case, I can change to a new emap.

Why is it important to fetch the base and emap out of the ThreadCache's copy? They are always just pointers to the same globals, no?